url47fcfc0133_baliosian_tau_fst_tgz-pJ8-uy_edu_fing_mina_fsa_test_old_TestComposition1J8/src/uy/edu/fing/mina/fsa/utils/Utils.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Created on 26-Jul-2004
+ *
+ * Copyright (C) 2004 Javier Baliosian
+ * All rights reserved.
+ * */
+package uy.edu.fing.mina.fsa.utils;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+
+import uy.edu.fing.mina.fsa.tf.TfString;
+import uy.edu.fing.mina.fsa.tffst.Tffst;
+import uy.edu.fing.mina.fsa.tffst.Transition;
+
+/**
+ * @author Javier Baliosian &lt; <a
+ *         href="mailto:jbaliosian@tsc.upc.es">jbaliosian@tsc.upc.es </a>&gt;
+ */
+public class Utils {
+
+  public static void showDot(String dot) {
+	String filename = "/tmp/tffst" + (new Random((new Date()).getTime())).nextInt(1000) + ".dot";
+	showDot(dot, filename);
+  }
+
+  public static void showDot(String dot, String filename) {
+	showDot(dot, filename, "png");  
+  }
+  
+  public static void showDot(String dot, String filename, String format) {
+
+	writeDot(dot, filename);
+
+	try {
+	  String cmd = "/usr/bin/dot -T" + format + " -o " + filename + "." + format + " " + filename;
+	  System.out.println("Calling Dot: " + cmd);
+	  Process p = Runtime.getRuntime().exec(cmd);
+	  p.waitFor();
+	  System.out.println("return: " + p.exitValue());
+	  p = Runtime.getRuntime().exec("eog " + filename + "." + format + " ");
+	} catch (IOException | InterruptedException e) {
+	  System.out.println("exception happened - here's what I know: ");
+	  e.printStackTrace();
+	  System.exit(-1);
+	}
+
+  }
+
+  public static void writeDot(String dot, String filename) {
+
+	try {
+	  BufferedWriter out = new BufferedWriter(new FileWriter(filename));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *out*. The resource is referenced at lines 58, 59. The resource is closed at line 60. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *out* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).